src/transformers/models/t5/modeling_t5.py

```diff
@@ -1595,12 +1595,12 @@ def forward(
 
         # If decoding with past key value states, only the last tokens
         # should be given as an input
-        if past_key_values is not None:
-            assert labels is None, "Decoder should not use cached key value states when training."
-            if decoder_input_ids is not None:
-                decoder_input_ids = decoder_input_ids[:, -1:]
-            if decoder_inputs_embeds is not None:
-                decoder_inputs_embeds = decoder_inputs_embeds[:, -1:]
+        # if past_key_values is not None and use_cache:
```

Comment:
It would be great if we could also delete the comments :-)

Reply:
OK,  the whole if statement and comments are deleted now.